pkg/cloud/cloud.go

```diff
@@ -28,17 +29,22 @@ type AWSKMS struct {
 }
 
 func New(region, kmsEndpoint string, qps, burst int) (*AWSKMS, error) {
+	sess, err := session.NewSession()
+	if err != nil {
+		return nil, fmt.Errorf("failed to create new session: %w", err)
+	}
+	if region == "" {
+		region, err = ec2metadata.New(sess).Region()
```

Comment:
Is this required? Doesn't the SDK do this for you if no region is specified?

Reply:
Yea, the region field won't be set on the session config, but try initiating a service like KMS and just listing keys without a region set.  
```go
package main

import (
	"encoding/json"
	"fmt"

	"github.com/aws/aws-sdk-go/aws/session"
	"github.com/aws/aws-sdk-go/service/kms"
)

func main() {
	svc := kms.New(session.Must(session.New()))
	result, err := svc.ListKeys(&kms.ListKeysInput{})
	if err != nil {
		panic(err)
	}
	data, err := json.MarshalIndent(result, "", "  ")
	if err != nil {
		panic(err)
	}
	fmt.Println(string(data))
}
```

Reply:
I tried to list with this code and it fails to list saying `MissingRegion: could not find region configuration`.

I also tried to run this [sample client](https://github.com/kubernetes-sigs/aws-encryption-provider/blob/master/cmd/client/main.go) it connects with the server, but fails to encrypt / decrypt if region is not set
```
2022/03/10 18:26:56 Failed to encrypt: rpc error: code = Unknown desc = failed to encrypt MissingRegion: could not find region configuration
```

Reply:
ok cool, TIL! I figured the region was automatically set